transport/filter.go

```diff
@@ -52,40 +52,54 @@ func ApplyFilter(o Outbound, f Filter) Outbound {
 	if f == nil {
 		return o
 	}
-	return filteredOutbound{o: o, f: f}
+	return filteredOutbound{Filter: f, Outbound: o}
 }
 
 // FilterFunc adapts a function into a Filter.
-type FilterFunc func(context.Context, *Request, Outbound) (*Response, error)
+type FilterFunc func(context.Context, *Request, RequestSender) (*Response, error)
 
-// Call for FilterFunc.
-func (f FilterFunc) Call(ctx context.Context, request *Request, out Outbound) (*Response, error) {
-	return f(ctx, request, out)
+// Send for FilterFunc.
+func (f FilterFunc) Send(ctx context.Context, req *Request, sender RequestSender) (*Response, error) {
+	return f(ctx, req, sender)
 }
 
 type filteredOutbound struct {
-	o Outbound
-	f Filter
+	Filter   Filter
+	Outbound Outbound
```

Comment:
WrappedOutbound?  Or NextOutbound?  Something that conveys that these are a chain of outbounds.